src/Makefile

```diff
@@ -37,13 +37,19 @@ YK_CFLAGS=	-DUSE_YK \
 YK_LDFLAGS=	`yk-config ${YK_BUILD_TYPE} --ldflags`
 YK_LIBS=	`yk-config ${YK_BUILD_TYPE} --libs`
 
-# If $(YK_DIR) is set then build with Yk JIT support.
+# If $(YK_BUILD_TYPE) is set then build with Yk JIT support.
 # The clang in $PATH must be the one from ykllvm.
-ifneq ($(strip $(YK_DIR)),)
+ifneq ($(strip $(YK_BUILD_TYPE)),)
 CC=		clang
 MYCFLAGS +=	$(YK_CFLAGS)
 MYLDFLAGS +=	$(YK_LDFLAGS)
 MYLIBS +=	$(YK_LIBS)
+
+check-ykconfig:
+	@which yk-config; \
+		if [ $$? -ne  0 ]; then \
+		echo "error: can't find yk-config in PATH"; exit 1; fi
+all: check-ykconfig
```

Comment:
`all` should probably be the first target, because otherwise it only triggers if you do `make all` (`make` executes the first target in a file if none is specified).

Reply:
It is the first target, but having revisiting this, I think the whole check is nonsense. The error message given is already fine.

Try f619841a1efe810819200e3e7ca35310eec40380